parse_snyk_output.cjs

```diff
@@ -0,0 +1,178 @@
+ 
+  console.log(inputData);
+
+  if (inputData) {
+    try {
+      const data = JSON.parse(inputData);
+      for (const project of data) {
+        vulnerabilities = vulnerabilities.concat(project.vulnerabilities);
+      }
+
+      console.log(vulnerabilities);
+
+    } catch (error) {
+      console.error('Error parsing Snyk output:', error);
+      vulnerabilities = parseNonJsonData(inputData);
+    }
+  }
+
+  return vulnerabilities;
+}
+
+
+function parseNonJsonData(inputData) {
+  let vulnerabilities = [];
+
+  // Custom logic to parse non-JSON inputData
+  const defaultTitle = 'Vulnerability Detected';
+
+  vulnerabilities.push({
+    title: defaultTitle,
+    description: `Non-JSON output from Snyk:\n\n${inputData}`
+  });
+
+  return vulnerabilities;
+}
+
+
+async function createJiraTicket(vulnerability) {
+   // DEFAULT DAYS  set to 60 adjust as needed
+   const sixtyDaysAgo = new Date(Date.now() - 60 * 24 * 60 * 60 * 1000).toISOString().split('T')[0];
+  //const today = new Date().toISOString().split('T')[0];
+
+  // JQL query to check ticket given title from the past given days
+  let jqlQuery = `project = "${process.env.JIRA_PROJECT_KEY}" AND summary ~ "MCR - ${vulnerability.title}" AND created >= ${sixtyDaysAgo}`;
+  let searchResult = await jira.searchJira(jqlQuery);
+
+
+  if (searchResult.issues && searchResult.issues.length > 0) {
+    for (const issue of searchResult.issues) {
+      if (issue.fields.status.name !== "Closed" && issue.fields.status.name !== "Cancelled") {
+        if (issue.fields.summary.startsWith('[MCR] -')) {
+          console.log(`Active Jira ticket already exists for vulnerability: ${vulnerability.title}`);
+          return;
+        }
+      }
+      
+    }
+  } else {
+                const issue = {
+                    fields: {
+                      project: {
+                        key: process.env.JIRA_PROJECT_KEY,
+                      },
+                      summary: `${process.env.JIRA_TITLE_PREFIX}  ${vulnerability.title}`,
+                      description: vulnerability.description,
+                      issuetype: {
+                        name: process.env.JIRA_ISSUE_TYPE,
+
+                      },
+                      labels: process.env.JIRA_LABELS.split(','),
+                      "customfield_10007" : process.env.JIRA_EPIC_KEY,
+                    },
+                  };
+
+                  
+
+                  let issueResponse;
+
+                  try {
+                    const jiraUrl = `${process.env.JIRA_BASE_URL}/rest/api/2/issue`;
+                    console.log('JIRA_URL:', jiraUrl);
+                    issueResponse = await jira.addNewIssue(issue);
+                    console.log(issueResponse)
+                    console.log(`Jira ticket created for vulnerability: ${vulnerability.title}`);
+                  } catch (error) {
+                    console.error('Error creating Jira ticket:', error);
+                  }
```

Comment:
Request: This try-catch is redundant, as is the try-catch around the attachment request below. This function is already being called from within a try-catch, so let's just let the exceptions bubble up to be logged in the caller.